packages/core/src/Config.cpp

```diff
@@ -707,15 +1128,229 @@ bool Config::configure_services(const SystemHandleInfoMap& info_map) const
+                  << topic_info_to.type << "] has been allowed by adding the following QoS policies: ";
+
+        auto policy_name = [&](xtypes::TypeConsistency to_check, const std::string& name) -> std::string
+            {
+              return (consistency & to_check) == to_check ? "'" + name + "' " : "";
+            };
+
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_TYPE_SIGN, "ignore type sign");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_TYPE_WIDTH, "ignore type width");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_SEQUENCE_BOUNDS, "ignore sequence bounds");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_ARRAY_BOUNDS, "ignore array bounds");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_STRING_BOUNDS, "ignore string bounds");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_MEMBER_NAMES, "ignore member names");
+        std::cout << policy_name(xtypes::TypeConsistency::IGNORE_MEMBERS, "ignore members");
+
+        std::cout << std::endl;
+      }
+    }
+  }
+
+  return valid;
+}
+
+//==============================================================================
+bool Config::check_service_compatibility(
+    const SystemHandleInfoMap& info_map,
+    const std::string& service_name,
+    const ServiceConfig& config) const
+{
+  bool valid = true;
+  for (const std::string& client : config.route.clients)
+  {
+    const auto it_client = info_map.find(client);
+    TopicInfo topic_info_client = remap_if_needed(client, config.remap,
+                                                  {service_name, config.request_type, config.reply_type});
+    const xtypes::DynamicType* client_type = resolve_type(it_client->second.types, topic_info_client.type);
+    /*
+    const auto client_type = it_client->second.types.find(topic_info_client.type);
+    if (client_type == it_client->second.types.end())
+    {
+      std::cerr << "Type [" << topic_info_client.type << "] not defined in middleware ["
+                << it_client->first << "]" << std::endl;
+      valid = false;
+      continue;
+    }
+    */
+
+    const auto it_server = info_map.find(config.route.server);
+    TopicInfo topic_info_server = remap_if_needed(config.route.server, config.remap,
+                                                  {service_name, config.request_type, config.reply_type});
+    const xtypes::DynamicType* server_type = resolve_type(it_server->second.types, topic_info_server.type);
+    /*
+    const auto server_type = it_server->second.types.find(topic_info_server.type);
+    if (server_type == it_server->second.types.end())
+    {
+      std::cerr << "Type [" << topic_info_server.type << "] not defined in middleware ["
+                << it_server->first << "]" << std::endl;
+      valid = false;
+      continue;
+    }
+    */
+
+    //if (client_type->second->is_compatible(*server_type->second) == xtypes::TypeConsistency::NONE)
```

Comment:
```suggestion

    const auto it_server = info_map.find(config.route.server);
    TopicInfo topic_info_server = remap_if_needed(config.route.server, 
                                                  config.remap,
                                                  {service_name, 
                                                   config.request_type,
                                                   config.reply_type});
    const xtypes::DynamicType* server_type = 
        resolve_type(it_server->second.types, topic_info_server.type);
```
Shall we remove the commented code?